projects/transactions-journey/src/lib/components/transaction-item/transaction-item.component.spec.ts

```diff
@@ -1,25 +1,69 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { Component } from '@angular/core';
+import { ComponentFixture, fakeAsync, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { AmountModule } from '@backbase/ui-ang';
 
 import { TransactionItemComponent } from './transaction-item.component';
 
+@Component({
+  selector: 'bb-host-component',
+  template: '<bb-transaction-item [transaction]="transaction"></bb-transaction-item>'
+})
+class HostComponent {
+  transaction = {
+    categoryCode: '#12a580',
+    dates: {
+      valueDate: 1600493600000
+    },
+    transaction: {
+      amountCurrency: {
+        amount: 5000,
+        currencyCode: 'EUR'
+      },
+      type: 'Salaries',
+      creditDebitIndicator: 'CRDT'
+    },
+    merchant: {
+      name: 'Backbase',
+      accountNumber: 'SI64397745065188826'
+    }
+  };
+}
+
 describe('TransactionItemComponent', () => {
-  let component: TransactionItemComponent;
-  let fixture: ComponentFixture<TransactionItemComponent>;
-
-  beforeEach(async () => {
-    await TestBed.configureTestingModule({
-      declarations: [ TransactionItemComponent ]
-    })
-    .compileComponents();
+  let component: HostComponent;
+  let fixture: ComponentFixture<HostComponent>;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [AmountModule],
+      declarations: [ TransactionItemComponent, HostComponent ]
+    });
   });
 
   beforeEach(() => {
-    fixture = TestBed.createComponent(TransactionItemComponent);
+    fixture = TestBed.createComponent(HostComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should set amount positive if the creditDebit indicator has the value of "credit"', async () => {
+    component.transaction.transaction.creditDebitIndicator = 'CRDT';
```

Comment:
I don't get what you are trying to mean. This is just setting the initial state of the test case to verify that it is setting the right class given the type of transaction

Reply:
1. Overall mutation is a bad practice, so I wanna try to avoid showing it even in the unit tests.
2. If you will agree on creating mocks, then this discussion is resolved (just use different mocks in different tests and no mutation will be needed, even the comment about `onPush` can be removed).
3. If you decide to go without mocks folder, then I would vote to merge these two lines in one. To avoid firstly creating the credit transaction and then assign it to a property. Hopes it makes more sense in this wording.

Reply:
after implementing mocks this test case changed a bit. Probably your concern does not apply anymore